cudapoa/src/allocate_block.hpp

```diff
@@ -70,11 +70,9 @@ class BatchBlock
             throw std::runtime_error(msg);
         }
 
-        // Calculate max POAs possible based on heirustics and available memory.
-        // TODO: Remove this fixed partitioing and making the filling up of buffer
-        // fully dynamic.
-        const float fraction_for_metadata = 0.4f;
-        max_poas_                         = (avail_mem * fraction_for_metadata) / device_size_per_poa;
+        // Calculate max POAs possible based on available memory.
+        int64_t device_size_per_score_matrix = (int64_t)matrix_sequence_dimension_ * (int64_t)max_graph_dimension_ * sizeof(ScoreT);
```

Comment:
`static_cast<int64_t>()` instead of old-style C cast like `(int64_t)`

Reply:
done.